coordinator/coordinator.go

```diff
@@ -22,13 +22,26 @@ func New() *Coordinator {
 // AddNode adds a new game node to the coordinator.
 func (c *Coordinator) AddNode(node *common.GameNode) {
 	c.nodesMutex.Lock()
-	defer c.nodesMutex.Unlock()
 	c.nodes[node.ID] = node
+	c.nodesMutex.Unlock()
```

Comment:
It's not a huge difference, but there is a slight performance overhead encountered using the `defer` keyword in golang. If it's a simple function, I think it's generally preferred to just call unlock after exiting the critical section.

Reply:
In the release notes for Go 1.14 I found this:

This release improves the performance of most uses of defer to incur almost zero overhead compared to calling the deferred function directly. As a result, defer can now be used in performance-critical code without overhead concerns. As a result, defer can now be used in performance-critical code without overhead concerns.

I think we shouldn't need to worry

Reply:
Nice, I was definitely operating on out-of-date knowledge